esp-hal/src/spi/master.rs

```diff
@@ -669,6 +687,8 @@ where
     ///
     /// Enables both input and output functionality for the pin, and connects it
     /// to the MISO signal and SIO1 input signal.
+    /// Use this for half-duplex SPI except for [DataMode::SingleTwoDataLines].
```

Comment:
I'll not block the PR on this, but the wording is a bit off. Users can use sio0/sio1 even if they intend to use SingleTwoWire, there may be devices on the bus that work in different modes.

Reply:
I had something in here about mixed busses? ... spooky